pkg/api/controllers/volumeGroup_test.go

```diff
@@ -102,3 +182,60 @@ func TestGetVolumeGroup(t *testing.T) {
 		assertTestResult(t, w.Code, 404)
 	})
 }
+
+func TestUpdateVolumeGroup(t *testing.T) {
+	var jsonStr = []byte(`{
+		"id": "3769855c-a102-11e7-b772-17b880d2f555",
+		"name": "volumeGroup-demo",
+  		"description": "volumeGroup test"
+	}`)
+
+	t.Run("Should return 202 if everything works well", func(t *testing.T) {
+		vg := model.VolumeGroupSpec{BaseModel: &model.BaseModel{
+			Id:        "3769855c-a102-11e7-b772-17b880d2f555",
+			UpdatedAt: time.Now().Format(constants.TimeFormat),
+		},
+			Status: "available",
+		}
+		json.NewDecoder(bytes.NewBuffer(jsonStr)).Decode(&vg)
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetVolumeGroup", c.NewAdminContext(), "3769855c-a102-11e7-b772-17b880d2f555").Return(&SampleVolumeGroups[1], nil)
+		mockClient.On("ListVolumesByGroupId", c.NewAdminContext(), SampleVolumeGroups[1].Id).Return(nil, nil)
+		mockClient.On("GetVolume", c.NewAdminContext(), "bd5b12a8-a101-11e7-941e-d77981b584d8").Return(&SampleVolumes[0], nil)
+		mockClient.On("UpdateVolumeGroup", c.NewAdminContext(), &vg).Return(&SampleVolumeGroups[1], nil)
+		db.C = mockClient
+
+		r, _ := http.NewRequest("PUT", "/v1beta/block/volumeGroups/3769855c-a102-11e7-b772-17b880d2f555", bytes.NewBuffer(jsonStr))
+		w := httptest.NewRecorder()
+		r.Header.Set("Content-Type", "application/JSON")
+		beego.InsertFilter("*", beego.BeforeExec, func(httpCtx *context.Context) {
+			httpCtx.Input.SetData("context", c.NewAdminContext())
+		})
+		beego.BeeApp.Handlers.ServeHTTP(w, r)
+		var output model.VolumeGroupSpec
+		json.Unmarshal(w.Body.Bytes(), &output)
+		assertTestResult(t, w.Code, 202)
+		assertTestResult(t, &output, &SampleVolumeGroups[1])
+	})
+
+	t.Run("Should return 500 if update volume with bad request", func(t *testing.T) {
```

Comment:
why 500?

Reply:
Will check once for bad request error code